Automattic/pocket-casts-android
# Transcripts - Search UI

## Description

This adds a search UI for Transcripts.
Internal discussion: p1723120902384739/1723091130.165579-slack-C078746V6UX

## Testing Instructions
1. Open transcript view for episode supporting transcripts e.g. ()
2. ✅ Notice search icon at the end of toolbar
3. Tap on search icon
4. ✅ Notice that a search bar is shown with an animation
5. Tap ">" arrow leading icon of the search bar
6. ✅ Notice that search bar closes with animation and search icon is visible
7. Tap on the search icon again
8. Tap close (x) icon while search bar is displaying
9. ✅ Notice that instead of screen closing, search bar is closed (this is done to prevent accidentally closing the screen while thinking that the close (x) button is to close the search bar)
10. Tap close (x) icon again while search bar is collapsed
11. ✅ Notice that the screen closes

## Screenshots or Screencast 

Portrait 

https://github.com/user-attachments/assets/870463c5-b75b-4d2c-9512-d55296011097

Landscape

https://github.com/user-attachments/assets/dbd54f11-0187-440a-aeac-a5cb6b5ca3f8

## Checklist
- [ ] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics. - **Analytics will be added as part of a separate task**
 
#### I have tested any UI changes...
